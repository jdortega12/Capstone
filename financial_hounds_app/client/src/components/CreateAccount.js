import React from 'react';
import "../styles/CreateAccount.css";
import { useState } from 'react';
import axios from 'axios';

const CreateAccount = () => {

    const [name, setName] = useState("");
    const [username, setUsername] = useState("");
    const [password, setPassword] = useState("");

    const handleCreate = async(e) => {
      const studentData = {"name": name, "username": username, "password": password};
      //alert(JSON.stringify(studentData));
      
      try{
        await axios({
          method: "post",
          url: "/createaccount",
          data: studentData,
        });
      } catch(error){
        console.log(error)
      }
        
    };

    return (
    <div className="myDiv">
      <form className="myForm" method='POST'>
        <h3 className="myLabel">Create Account</h3>
        <div className="mb-3">
          <label className="myLabel">Full name</label>
          <input
            name="name"
            type="text"
            value = {name}
            onChange = {(e) => setName(e.target.value)}
            className="form-control"
            placeholder="Full name"
          />
        </div>
        <div className="mb-3">
          <label className="myLabel">Email</label>
          <input
          name="username"
          type="text" 
          value = {username}
          onChange = {(e) => setUsername(e.target.value)}
          className="form-control" 
          placeholder="Enter email" />
        </div>
        <div className="mb-3">
          <label className="myLabel">Password</label>
          <input
            name="password"
            value = {password}
            onChange = {(e) => setPassword(e.target.value)}
            type="password"
            className="form-control"
            placeholder="Enter password"
          />
        </div>
        <div className="d-grid">
          <button onClick={()=>handleCreate()} className="myButton">
            Submit
          </button>
        </div>
      </form>

    <div>
      <p className="paragraphCreateAccount">
            Disclaimer: Greyhound Financial Services will only use your inputted financial data for budget and financial planning
            calculations.
      </p>
    </div>
    </div>
    );
}

export default CreateAccount;