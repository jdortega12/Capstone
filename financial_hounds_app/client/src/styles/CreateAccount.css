.myDiv{
    background-color: #0d4b34;
    height: 100vh;
}

.myForm{
    justify-content: center;
    align-items: center;
    height: auto;
    max-width: 30%;
    margin-left: auto;
    margin-right: auto;
}

.myLabel{
    color: white;
}

.myButton{
  margin-left: 60px;
  border: none;
  border-radius: 5px;
  cursor: pointer;
  justify-content: center;
  background-color: #282c34;
  height: 50px;
  width: 180px;
  color: white;
  text-decoration: none;
  font-size: 20px;
}

button:hover {
    background-color: #07090a;
    cursor: pointer;
    transition: 0.3s ease-in;
  }

.paragraphCreateAccount{
    text-align: center;
    border: 3px solid;
    border-color: white;
    margin-left: auto;
    margin-right: auto;
    font-size: 20px;
    font-weight: lighter;
    color: white;
    width: 33.3%;
    margin-top: 16px;
}