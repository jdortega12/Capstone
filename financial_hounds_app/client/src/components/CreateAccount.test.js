import React from 'react';
import CreateAccount from '../components/CreateAccount';
import renderer from 'react-test-renderer';

test("Create Account Snapshot Test", ()=> {
    const component = renderer.create(
        <CreateAccount />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});