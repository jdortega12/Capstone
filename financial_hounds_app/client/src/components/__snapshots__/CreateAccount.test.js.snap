// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Account Snapshot Test 1`] = `
<div
  className="myDiv"
>
  <form
    className="myForm"
    method="POST"
  >
    <h3
      className="myLabel"
    >
      Create Account
    </h3>
    <div
      className="mb-3"
    >
      <label
        className="myLabel"
      >
        Full name
      </label>
      <input
        className="form-control"
        name="name"
        onChange={[Function]}
        placeholder="Full name"
        type="text"
        value=""
      />
    </div>
    <div
      className="mb-3"
    >
      <label
        className="myLabel"
      >
        Username
      </label>
      <input
        className="form-control"
        name="username"
        onChange={[Function]}
        placeholder="Enter username"
        type="text"
        value=""
      />
    </div>
    <div
      className="mb-3"
    >
      <label
        className="myLabel"
      >
        Password
      </label>
      <input
        className="form-control"
        name="password"
        onChange={[Function]}
        placeholder="Enter password"
        type="password"
        value=""
      />
    </div>
    <div
      className="d-grid"
    >
      <button
        className="myButton"
        onClick={[Function]}
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
